### DISCUSSION TITLE
Convert TcUnit to Bosch Rexroth (IndraWorks) and verify with TcUnit-Verifier

### DISCUSSION THREAD

**Discussion by Honeycakehorse** [tcunit/TcUnit#247] [Q&A] [answered]
Hello there,

We would like to use TcUnit for our programs that we develop on the Bosch Rexroth platform. Although the programs of this platform are also written with CoDeSys, we lack the special Beckhoff libraries that TcUnit accesses. We have therefore reproduced these functions or use libraries from Bosch Rexroth.  
Now we want to verify our converted TcUnit. For this we want to use 'TcUnit-Verifier'. Unfortunately, we can only get the 'TcUnit-Verifier_TwinCAT' to run. Unfortunately, 'TcUnit-Verifier_DotNet' does not work because Bosch Rexroth does not support Visual Studio.

It is therefore necessary to perform a manual check of all outputs (equal or different) from TcVT. But what is the best way to do this? Well, some tests fail and some do not. Is there any information about which tests should fail?

**Comment by I-Campbell** [comment, marked as answer]
There is a codesys native fork, [counit](https://forge.codesys.com/lib/counit/home/Home/) , which includes a conversion of the verifier.

**Comment by Honeycakehorse** [comment]
Thanks [I-Campbell](https://github.com/I-Campbell) for the tip, I didn't know that. We have now created a fork from TcUnit ourselves. It works so far, but certainly still has many bugs.

We could now use CoUnit, but it is already three years old. Then we use IndraWorks from Bosch Rexroth and not the CoDeSys programming environment directly. We still have to make sure that CoUnit also runs there.

Our version of TCUnit rewritten for CoDeSys already writes all errors of the 'TcUnit-Verifier_TwinCAT' in an XML file. Is it possible to test this XML file externally with a C# program or similar without TcUnit running on VisualStudio?
